### ISSUE TITLE
🐛  user-provided comparison function does not correctly implement a total order

### ISSUE THREAD

**Issue by AndreiTS** [biomejs/biome#6536]
### Environment information

```block
CLI:
  Version:                      2.0.5
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v24.2.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.12.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

Running biome in CLI or local in my project gives the following error:

```
Biome encountered an unexpected error
This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.
When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:
Source Location: library/core/src/slice/sort/shared/smallsort.rs:860:5
Thread Name: biome::workspace_worker_0
Message: user-provided comparison function does not correctly implement a total order
```

Biome doesn't provide the code that is causing this error. After this message it continues showing lint errors:

![Image](https://github.com/user-attachments/assets/a6bd8277-5407-4c17-a1f1-aedd07e74da9)

### Expected result

I expect no errors.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by AndreiTS** [comment]
[biome.json](https://github.com/user-attachments/files/20889685/biome.json)

**Comment by ematipico** [comment]
Can you provide also with the code that causes the issue? Without it, we can't fix the bug.

**Comment by AndreiTS** [comment]
@ematipico I can't because biome doesn't tell me what code is causing this. Is there a verbose mode maybe?

**Comment by ematipico** [comment]
Yes, and it's called `--verbose`. I know Biome doesn't tell you the file, that's why we're asking to do that 😃

**Comment by AndreiTS** [comment]
This is the output
![Image](https://github.com/user-attachments/assets/812fe881-cd99-4b4f-b1f4-2a94b77f7ae1)

**Comment by ematipico** [comment]
@AndreiTS, sorry if I wasn't clear. I am asking if you can identify the user file that causes the error. There should be some JSON/CSS/JS file that causes this. Since I don't know your configuration, I can't understand which rule is causing this (we have a bunch of ordering rules).

Alternatively, provide us with a reproduction where you can reproduce the issue. It must be **minimal**, if possible.

**Comment by AndreiTS** [comment]
I cloned biome and added Backtrace to the panic handler:

```
Backtrace: Backtrace [
    { fn: "std::backtrace_rs::backtrace::win64::trace", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\..\..\backtrace\src\backtrace\win64.rs", line: 85 },
    { fn: "std::backtrace_rs::backtrace::trace_unsynchronized", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\..\..\backtrace\src\backtrace\mod.rs", line: 66 },   
    { fn: "std::backtrace::Backtrace::create", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\backtrace.rs", line: 331 },
    { fn: "std::backtrace::Backtrace::force_capture", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\backtrace.rs", line: 312 },
    { fn: "biome_cli::panic::panic_handler", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_cli\src\panic.rs", line: 12 },
    { fn: "core::ops::function::Fn::call<void (*)(ref$<std::panic::PanicHookInfo>),tuple$<ref$<std::panic::PanicHookInfo> > >", file: "C:\Users\andrei\.rustup\toolchains\1.87.0-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs", line: 79 },
    { fn: "std::panicking::rust_panic_with_hook", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\panicking.rs", line: 841 },
    { fn: "std::panicking::begin_panic_handler::closure$0", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\panicking.rs", line: 699 },
    { fn: "std::sys::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\sys\backtrace.rs", line: 168 },
    { fn: "std::panicking::begin_panic_handler", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\panicking.rs", line: 697 },
    { fn: "core::panicking::panic_fmt", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\core\src\panicking.rs", line: 75 },
    { fn: "core::slice::sort::shared::smallsort::panic_on_ord_violation", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\core\src\slice\sort\shared\smallsort.rs", line: 860 },
    { fn: "core::slice::sort::shared::smallsort::bidirectional_merge<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember,bool (*)(ref$<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember>,ref$<biome_js_analyze::assist::source::use_sorted_keys:", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\sort\shared\smallsort.rs", line: 838 },
    { fn: "core::slice::sort::shared::smallsort::small_sort_general_with_scratch<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember,bool (*)(ref$<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember>,ref$<biome_js_analyze::assist::source::use_", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\sort\shared\smallsort.rs", line: 286 },
    { fn: "core::slice::sort::shared::smallsort::small_sort_general<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember,bool (*)(ref$<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember>,ref$<biome_js_analyze::assist::source::use_sorted_keys::", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\sort\shared\smallsort.rs", line: 217 },
    { fn: "core::slice::sort::shared::smallsort::impl$4::small_sort", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\sort\shared\smallsort.rs", line: 129 },
    { fn: "core::slice::sort::shared::smallsort::impl$3::small_sort", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\sort\shared\smallsort.rs", line: 101 }, 
    { fn: "core::slice::sort::unstable::quicksort::quicksort<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember,bool (*)(ref$<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember>,ref$<biome_js_analyze::assist::source::use_sorted_keys::ObjectM", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\sort\unstable\quicksort.rs", line: 31 },
    { fn: "core::slice::sort::unstable::ipnsort<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember,bool (*)(ref$<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember>,ref$<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember>)>", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\sort\unstable\mod.rs", line: 84 },
    { fn: "core::slice::impl$0::sort_unstable<biome_js_analyze::assist::source::use_sorted_keys::ObjectMember>", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\slice\mod.rs", line: 2982 },
    { fn: "biome_js_analyze::assist::source::use_sorted_keys::impl$0::action", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_js_analyze\src\assist\source\use_sorted_keys.rs", 
line: 162 },
    { fn: "biome_analyze::signals::impl$24::actions<enum2$<biome_js_analyze::assist::source::use_sorted_keys::UseSortedKeys> >", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_analyze\src\signals.rs", line: 412 },
    { fn: "biome_service::file_handlers::ProcessLint::process_signal<biome_js_syntax::syntax_node::JsLanguage>", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_service\src\file_handlers\mod.rs", line: 557 },
    { fn: "biome_service::file_handlers::javascript::lint::closure$0", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_service\src\file_handlers\javascript.rs", line: 788 },    
    { fn: "core::ops::function::impls::impl$3::call_mut<tuple$<ref$<dyn$<biome_analyze::signals::AnalyzerSignal<biome_js_syntax::syntax_node::JsLanguage> > > >,dyn$<core::ops::function::FnMut<tuple$<ref$<dyn$<biome_analyze::signals::AnalyzerSignal<biome_js_syntax::sy", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\ops\function.rs", line: 294 },
    { fn: "biome_analyze::PhaseRunner<biome_js_syntax::syntax_node::JsLanguage,biome_analyze::matcher::InspectMatcher<biome_js_analyze::analyze::closure_env$0<biome_service::file_handlers::javascript::lint::closure_env$0,enum2$<biome_analyze::Never> >,biome_analyze::", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_analyze\src\lib.rs", line: 568 },
    { fn: "biome_analyze::PhaseRunner<biome_js_syntax::syntax_node::JsLanguage,biome_analyze::matcher::InspectMatcher<biome_js_analyze::analyze::closure_env$0<biome_service::file_handlers::javascript::lint::closure_env$0,enum2$<biome_analyze::Never> >,biome_analyze::", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_analyze\src\lib.rs", line: 398 },
    { fn: "biome_analyze::Analyzer<biome_js_syntax::syntax_node::JsLanguage,biome_analyze::matcher::InspectMatcher<biome_js_analyze::analyze::closure_env$0<biome_service::file_handlers::javascript::lint::closure_env$0,enum2$<biome_analyze::Never> >,biome_analyze::reg", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_analyze\src\lib.rs", line: 180 },
    { fn: "biome_js_analyze::analyze_with_inspect_matcher<biome_js_analyze::analyze::closure_env$0<biome_service::file_handlers::javascript::lint::closure_env$0,enum2$<biome_analyze::Never> >,biome_service::file_handlers::javascript::lint::closure_env$0,enum2$<biome_", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_js_analyze\src\lib.rs", line: 173 },
    { fn: "biome_js_analyze::analyze<biome_service::file_handlers::javascript::lint::closure_env$0,enum2$<biome_analyze::Never> >", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_js_analyze\src\lib.rs", line: 198 },
    { fn: "biome_service::file_handlers::javascript::lint", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_service\src\file_handlers\javascript.rs", line: 782 },
    { fn: "biome_service::workspace::server::impl$2::pull_diagnostics", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_service\src\workspace\server.rs", line: 1376 },
    { fn: "biome_service::workspace::FileGuard<dyn$<biome_service::workspace::Workspace> >::pull_diagnostics<dyn$<biome_service::workspace::Workspace> >", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_service\src\workspace.rs", line: 1443 },
    { fn: "biome_cli::execute::process_file::lint_and_assist::analyze_with_guard", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_cli\src\execute\process_file\lint_and_assist.rs", line: 113 },
    { fn: "biome_cli::execute::process_file::check::check_file", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_cli\src\execute\process_file\check.rs", line: 38 },
    { fn: "biome_cli::execute::process_file::process_file", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_cli\src\execute\process_file.rs", line: 232 },
    { fn: "biome_cli::execute::traverse::handle_file::closure$0", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_cli\src\execute\traverse.rs", line: 644 },
    { fn: "std::panicking::try::do_call<biome_cli::execute::traverse::handle_file::closure_env$0,enum2$<core::result::Result<enum2$<biome_cli::execute::process_file::FileStatus>,enum2$<biome_cli::execute::process_file::Message> > > >", file: "C:\Users\andrei\.rustup\toolchains\1.87.0-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs", line: 589 },    
    { fn: "std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<tokio::runtime::task::harness::poll_future::closure_env$1<tracing::instrument::Instrumented<enum2$<biome_cli::service::windows::run_server::async_fn_env$0> >,alloc::sync::Arc<tokio::runtim" },
    { fn: "std::panicking::try", file: "C:\Users\andrei\.rustup\toolchains\1.87.0-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs", line: 552 },
    { fn: "std::panic::catch_unwind<biome_cli::execute::traverse::handle_file::closure_env$0,enum2$<core::result::Result<enum2$<biome_cli::execute::process_file::FileStatus>,enum2$<biome_cli::execute::process_file::Message> > > >", file: "C:\Users\andrei\.rustup\toolchains\1.87.0-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs", line: 359 },
    { fn: "biome_cli::execute::traverse::handle_file", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_cli\src\execute\traverse.rs", line: 644 },
    { fn: "biome_cli::execute::traverse::impl$2::handle_path", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_cli\src\execute\traverse.rs", line: 629 },
    { fn: "biome_fs::fs::os::impl$5::handle::closure$0", file: "C:\Users\andrei\Documents\GitHub\biome\crates\biome_fs\src\fs\os.rs", line: 220 },
    { fn: "rayon_core::scope::impl$0::spawn::closure$0::closure$0<biome_fs::fs::os::impl$5::handle::closure_env$0>", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\scope\mod.rs", line: 526 },
    { fn: "core::panic::unwind_safe::impl$25::call_once<tuple$<>,rayon_core::scope::impl$0::spawn::closure$0::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\panic\unwind_safe.rs", line: 272 },
    { fn: "std::panicking::try::do_call<core::panic::unwind_safe::AssertUnwindSafe<rayon_core::scope::impl$0::spawn::closure$0::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0> >,tuple$<> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\panicking.rs", line: 589 },
    { fn: "std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<rayon_core::scope::impl$0::spawn::closure$0::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0> >,tuple$<> >" },
    { fn: "std::panicking::try", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\panicking.rs", line: 552 },
    { fn: "std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<rayon_core::scope::impl$0::spawn::closure$0::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0> >,tuple$<> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\panic.rs", line: 359 },
    { fn: "rayon_core::unwind::halt_unwinding<rayon_core::scope::impl$0::spawn::closure$0::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0>,tuple$<> >", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\unwind.rs", line: 17 },
    { fn: "rayon_core::scope::ScopeBase::execute_job_closure<rayon_core::scope::impl$0::spawn::closure$0::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0>,tuple$<> >", file: 
"C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\scope\mod.rs", line: 689 },
    { fn: "rayon_core::scope::ScopeBase::execute_job<rayon_core::scope::impl$0::spawn::closure$0::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0> >", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\scope\mod.rs", line: 679 },
    { fn: "rayon_core::scope::impl$0::spawn::closure$0<biome_fs::fs::os::impl$5::handle::closure_env$0>", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\scope\mod.rs", line: 526 },
    { fn: "rayon_core::job::impl$6::execute<rayon_core::scope::impl$0::spawn::closure_env$0<biome_fs::fs::os::impl$5::handle::closure_env$0> >", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\job.rs", line: 169 },
    { fn: "rayon_core::job::JobRef::execute", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\job.rs", line: 64 },
    { fn: "rayon_core::registry::WorkerThread::execute", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\registry.rs", line: 860 },       
    { fn: "rayon_core::registry::WorkerThread::wait_until_cold", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\registry.rs", line: 794 },
    { fn: "rayon_core::registry::WorkerThread::wait_until<rayon_core::latch::OnceLatch>", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\registry.rs", line: 769 },
    { fn: "rayon_core::registry::WorkerThread::wait_until_out_of_work", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\registry.rs", line: 818 },
    { fn: "rayon_core::registry::main_loop", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\registry.rs", line: 923 },
    { fn: "rayon_core::registry::ThreadBuilder::run", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\registry.rs", line: 53 },
    { fn: "rayon_core::registry::impl$2::spawn::closure$0", file: "C:\Users\andrei\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rayon-core-1.12.1\src\registry.rs", line: 98 },     
    { fn: "core::hint::black_box", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\hint.rs", line: 477 },
    { fn: "std::sys::backtrace::__rust_begin_short_backtrace<rayon_core::registry::impl$2::spawn::closure_env$0,tuple$<> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\sys\backtrace.rs", line: 152 },
    { fn: "std::thread::impl$0::spawn_unchecked_::closure$1::closure$0<rayon_core::registry::impl$2::spawn::closure_env$0,tuple$<> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\thread\mod.rs", line: 559 },
    { fn: "core::panic::unwind_safe::impl$25::call_once<tuple$<>,std::thread::impl$0::spawn_unchecked_::closure$1::closure_env$0<rayon_core::registry::impl$2::spawn::closure_env$0,tuple$<> > >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\panic\unwind_safe.rs", line: 272 },
    { fn: "std::panicking::try::do_call<core::panic::unwind_safe::AssertUnwindSafe<std::thread::impl$0::spawn_unchecked_::closure$1::closure_env$0<rayon_core::registry::impl$2::spawn::closure_env$0,tuple$<> > >,tuple$<> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\panicking.rs", line: 589 },
    { fn: "rayon_core::unwind::impl$0::drop" },
    { fn: "std::panicking::try", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\panicking.rs", line: 552 },
    { fn: "std::panic::catch_unwind", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\panic.rs", line: 359 },
    { fn: "std::thread::impl$0::spawn_unchecked_::closure$1<rayon_core::registry::impl$2::spawn::closure_env$0,tuple$<> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\std\src\thread\mod.rs", line: 557 },
    { fn: "core::ops::function::FnOnce::call_once<std::thread::impl$0::spawn_unchecked_::closure_env$1<rayon_core::registry::impl$2::spawn::closure_env$0,tuple$<> >,tuple$<> >", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359\library\core\src\ops\function.rs", line: 250 },
    { fn: "alloc::boxed::impl$28::call_once", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\alloc\src\boxed.rs", line: 1966 },
    { fn: "alloc::boxed::impl$28::call_once", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\alloc\src\boxed.rs", line: 1966 },
    { fn: "std::sys::pal::windows::thread::impl$0::new::thread_start", file: "/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library\std\src\sys\pal\windows\thread.rs", line: 56 },       
    { fn: "BaseThreadInitThunk" },
    { fn: "RtlUserThreadStart" },
]
```

The rule causing the issue is https://biomejs.dev/assist/actions/use-sorted-keys/
After this, I added logging to the function that was causing the panic and found the source.

This is the minimal example:

```
const newObj = {
    ...obj,
    a: '',
    b: '',
    c: '',
    d: '',
    e: '',
    f: '',
    g: '',
    h: '',
    i: '',
    j: '',
    k: '',
    l: '',
    m: '',
    n: '',
    o: '',
    p: '',
    q: '',
    r: '',
    s: '',
    t: '',
    u: '',
    v: '',
    w: '',
    x: '',
    z: '',
    aa: '',
    ab: '',
    ac: '',
    ad: '',
    ae: '',
    af: '',
    ag: '',
};
```

The original code is more complex.
If you remove any key it will work.
Run with `biome check --formatter-enabled=false --linter-enabled=false`

If you need my config:
[biome.json](https://github.com/user-attachments/files/20903797/biome.json)

![Image](https://github.com/user-attachments/assets/b2d2b431-0a26-4016-bc0d-aacc6d2797a4)

**Comment by ematipico** [comment]
Hey @Conaclos, it seems this bug is very similar to what we have in the JSON files.
